test/lib.js

```diff
@@ -218,6 +218,20 @@ test('createUrl 10', t => {
   t.end();
 });
 
+test('download unique filename', async t => {
+  const options = {
+    dir: __dirname
+  };
+  const url = 'https://source.unsplash.com/WLUHO9A_xik/1600x900';
+  const expectedFilename = __dirname+'/'+'wallpaper-uploads-141310026617203b5980d-c86b8baa.jpg';
+  const stateStack = [];
+
+  const file = await download(options, url, state => stateStack.push(state));
```

Comment:
I understand why you added the stateStack function here, I tried to remove this from the previous test but it is expecting a function for that prop. I've made that parameter optional now so you are free to remove the third param for `download`, you'll just need to rebase/merge in master first.